components/script/dom/filereader.rs

```diff
@@ -245,7 +272,28 @@ impl FileReader {
             format!("data:{};base64,{}", data.blobtype, base64)
         };
 
-        DOMString::from(output)
+        StringOrObject::String(DOMString::from(output))
+    }
+
+    // https://w3c.github.io/FileAPI/#dfn-readAsArrayBuffer
+    #[allow(unsafe_code, unused_variables)]
+    fn perform_readasarraybuffer(cx: *mut JSContext, data: ReadMetaData, bytes: &[u8])
+        -> StringOrObject {
+        unsafe {
+            let length = bytes.len() as u32;
+            let newArrayBuffer = JS_NewArrayBuffer(cx, length);
```

Comment:
Both `FileAPI/reading-data-section/filereader_readAsArrayBuffer.html` and `FileAPI/reading-data-section/filereader_result.html` crash at this point. There's very little information in the log to debug it, it only shows a stack trace. I still have to test for additional tests which have been mentioned in the issue.